statement.k

```diff
@@ -44,8 +44,108 @@ rule
     <k> #exeStmt(S:SimpleStatement;) => S; ...</k>
     <callStack> .List </callStack>
 
-rule
-    <k> #localFunctionCall(FN:Id, ARGS:Values) => functionCall(FN; ARGS) ... </k>     [structural]
+syntax ContractCellMap
+syntax Map ::= ContractCellMap
+
+
+//**************************************************************************//
+// I was planing to have the following rule for address conversion.
+// However, the behavior of the in_keys() funciton is not what I expected.
+// Thus, I have another work-around, which works, but may not be perfect.
+// Maybe someone can help me to check on this?
+
+/*
+rule [Address-Conversion]:
+    <k> #localFunctionCall(FN:Id, V:Value, .Values) => V ... </k>
+    <contracts> CTS </contracts>
+    requires (FN in_keys(CTS))
+*/
+
+rule [Address-Conversion]:
+    <k> #localFunctionCall(FN:Id, V:Value, .Values) => V ... </k>
+    <Contract>
+        <cname> FN </cname>
+        ...
+    </Contract>
+
+//******************************************************************/
+
+
+//**************************************************************************//
+// I was planing to have the following rule for Syntactic-Sugar-Internal-Call.
+// However, the behavior of the in_keys() funciton is not what I expected.
+// Thus, I have another work-around, which works, but may not be perfect.
+// Maybe someone can help me to check on this?
+
+/*
+rule [Syntactic-Sugar-Internal-Call]:
+    <k> #localFunctionCall(FN:Id, ARGS:Values) => functionCall(FN; ARGS) ... </k>
+    <contracts> CTS </contracts>
+    requires notBool (FN in keys(CTS))
+*/
```

Comment:
We should use @dwightguth 's suggestion if it works here:

```
syntax Bool ::= KItem "in_keys" "(" ContractCellMap ")" [function, hook(MAP.in_keys)]
```

Reply:
I have tried this approach. However, during the compilation, there are warnings about ambiguity. And, the result is not correct.

Reply:
Use `syntax Bool ::= Id "in_contracts" "(" ContractCellMap ")"` to avoid the parsing ambiguity.

Reply:
Hi, @ehildenb, I have revised it as 
```
syntax Bool ::= Id "in_contracts" "(" ContractCellMap ")" [function, hook(MAP.in_keys)]
```
Now, there is no parsing ambiguity, but it still does not work. It seems that it returns false, even the `Id` is in the `ContractCellMap`. I have pushed the change on GitHub. Could you help me to check the reason why it does not work? You can alternate the following two rules to check:

```
/*    line 58 in statement.k
rule [Address-Conversion]:
    <k> #localFunctionCall(FN:Id, (V:Value, .Values)) => V ... </k>
    <contracts> CTS </contracts>
    requires (FN in_contracts(CTS))
*/

rule [Address-Conversion]:
    <k> #localFunctionCall(FN:Id, (V:Value, .Values)) => V ... </k>
    <Contract>
        <cname> FN </cname>
        ...
    </Contract>``